Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Route.hs

```diff
@@ -18,24 +18,54 @@ module Domain.Action.UI.Route
     getRoutes,
     getPickupRoutes,
     getTripRoutes,
+    GetPickupRoutesReq,
   )
 where
 
 import qualified Domain.Types.Merchant as Merchant
 import qualified Domain.Types.Person as DP
+import Domain.Types.Ride
+import Environment
 import Kernel.External.Types (ServiceFlow)
 import Kernel.Prelude
 import Kernel.Storage.Esqueleto.Config (EsqDBReplicaFlow)
 import Kernel.Types.Id
+import Kernel.Utils.Common (fromMaybeM)
+import qualified Storage.CachedQueries.Merchant.MerchantServiceUsageConfig as QMSUC
+import qualified Storage.Queries.Ride as QRide
+import Tools.Error
 import qualified Tools.Maps as Maps
 
+data GetPickupRoutesReq = GetPickupRoutesReq
+  { waypoints :: NonEmpty Maps.LatLong,
+    mode :: Maybe Maps.TravelMode,
+    calcPoints :: Bool,
+    rideId :: Maybe (Id Ride)
+  }
+  deriving (Generic, ToJSON, FromJSON, Show, ToSchema)
+
 getRoutes :: (ServiceFlow m r, EsqDBReplicaFlow m r) => (Id DP.Person, Id Merchant.Merchant) -> Maps.GetRoutesReq -> m Maps.GetRoutesResp
 getRoutes (personId, merchantId) req = do
   Maps.getRoutes Nothing personId merchantId Nothing req
 
-getPickupRoutes :: ServiceFlow m r => (Id DP.Person, Id Merchant.Merchant) -> Maps.GetRoutesReq -> m Maps.GetRoutesResp
-getPickupRoutes (personId, merchantId) req = do
-  Maps.getPickupRoutes personId merchantId Nothing req
+getPickupRoutes :: (Id DP.Person, Id Merchant.Merchant) -> GetPickupRoutesReq -> Flow Maps.GetRoutesResp
+getPickupRoutes (personId, merchantId) GetPickupRoutesReq {..} = do
+  mocId <- Maps.getMerchantOperatingCityId personId Nothing
```

Comment:
Can runWithServiceConfig function be used here with (.getPickupRoutes) or (.getFirstPickupRoute) changed accordingly ?

Reply:
Same as above